### PR TITLE
Core: Simplified hierarchy separators

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #8793 

## What I did

- Add warnings when separators are set.
- Change behaviour when hierarchy separators are not set, as per issue.
- Add showRoots option and change behaviour when set.

## How to test

- Added a test for `stories.js` and new behaviour -- should I check the deprecation message is called?

- Run official storybook to see deprecation message when kinds still have '/' or '.'
- Run other examples to see deprecation message when option is set
- Run `examples/cra-react15` to see new behaviour (try setting `showRoots: true`)

- Does this need an update to the documentation? Yes, I removed documentation about the old option, added documentation for the new option, and migration notes.

If your answer is yes to any of these, please make sure to include it in your PR.

## Still TODO (on this PR or later)

 - [x] Can we add a test for the docs part? @shilman can you help me here?
 - [x] Add a codemod to convert `.` and `|` to `/` in kinds.
 - [x] Update all our example apps to use the new API
 - [ ] Check all our documentation for uses of `|` or `.`

### CODE REVIEW

File: examples/cra-react15/.storybook/config.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const loadFn = () => {
   require('../src/stories/welcome.stories');
 
   // test mixtures of storiesof & module files
-  const req = require.context('../src/stories', true, /\.stories\.js$/);
+  const req = require.context('../src/stories', true, /button\.stories\.js$/);
```
Comment (tmeasday) [inline examples/cra-react15/.storybook/config.js]: This wasn't working before

File: lib/client-api/src/client_api.ts [hunk 2]
```diff
@@ -129,6 +129,7 @@ export default class ClientApi {
     };
   };
 
+  // FIXME: not sure what to do about this API
```
Comment (tmeasday) [inline lib/client-api/src/client_api.ts]: I guess this is a public API. Should I do something like:

```js
if (/* separators are set */) {
  return separators;
} else if (/* any kind has a - or a | and showRoots is undefined */) {
  return oldDefaults;
} else {
  return { hierarchySeparator: '/' };
}
```

File: examples/cra-react15/.storybook/config.js (hunk 1 above)
Comment (shilman) [reply examples/cra-react15/.storybook/config.js]: 🙈

File: lib/client-api/src/client_api.ts (hunk 2 above)
Comment (shilman) [reply lib/client-api/src/client_api.ts]: That makes sense to me

**Comment by shilman** [comment]
This looks great @tmeasday. I'll see if I can add a test for the docs part! 👍

**Review by mrmckeb — APPROVED** [review]

**Comment by shilman** [comment]
@tmeasday mind taking a look at the failing tests?

**Review by shilman — APPROVED** [review]
LGTM

**Pull request merged**